packages/graphql-yoga/src/plugins/useSchemaDef.ts

```diff
@@ -0,0 +1,133 @@
+import { makeExecutableSchema } from '@graphql-tools/schema'
+import { IResolvers, TypeSource } from '@graphql-tools/utils'
+import { GraphQLError, GraphQLSchema, isSchema } from 'graphql'
+import { Plugin, PromiseOrValue, YogaInitialContext } from 'graphql-yoga'
+
+// TODO: Will be removed later
+type TypeDefsAndResolvers<TContext, TRootValue = {}> = {
+  typeDefs: TypeSource
+  resolvers?:
+    | IResolvers<TRootValue, TContext>
+    | Array<IResolvers<TRootValue, TContext>>
+}
+
+export type YogaSchemaDefinition<TContext, TRootValue> =
+  | TypeDefsAndResolvers<TContext, TRootValue>
+  | PromiseOrValue<GraphQLSchema>
+  | ((request: Request) => PromiseOrValue<GraphQLSchema>)
+
+// Will be moved to a seperate export later
+export function getDefaultSchema() {
+  return makeExecutableSchema({
+    typeDefs: /* GraphQL */ `
+      """
+      Greetings from GraphQL Yoga!
+      """
+      type Query {
+        greetings: String
+      }
+      type Subscription {
+        """
+        Current Time
+        """
+        time: String
+      }
+    `,
+    resolvers: {
+      Query: {
+        greetings: () =>
+          'This is the `greetings` field of the root `Query` type',
+      },
+      Subscription: {
+        time: {
+          async *subscribe() {
+            while (true) {
+              yield { time: new Date().toISOString() }
+              await new Promise((resolve) => setTimeout(resolve, 1000))
+            }
+          },
+        },
+      },
+    },
+  })
+}
+
+export const useSchemaDef = <
```

Comment:
What does the Def in `useSchemaDef` stand for?

Could we just do `useSchema` instead?

Reply:
I want to avoid the confusion with Envelop's useSchema. Also we still do support an object of typeDefs and resolvers. I just wanted to distinguish the purposes.

Reply:
I think it can do the same as Envelop `useSchema` (and more), thus why not remove export of envelop/core useSchema and replace it with this? 🤔

Reply:
Done!